File: 826. Most Profit Assigning Work/Solution.go
type store struct{
    diff int
    prof int
}

func maxProfitAssignment(difficulty []int, profit []int, worker []int) int {
    sort.Slice(worker, func(i,j int) bool {
        return worker[i] > worker[j]
    })
    value := []store{}
    for i:=0;i<len(profit);i++ {
        value = append(value, store{
            diff:difficulty[i], 
            prof:profit[i],
        })
    }
    sort.Slice(value, func (i,j int) bool {
        return value[i].prof > value[j].prof
    })
    max_profit := 0
    idx := 0
    for _,w := range worker{
        for idx < len(value) {
            if w >= value[idx].diff {
                max_profit += value[idx].prof
                break
            }
            idx += 1
        }
    }

    return max_profit
}

File: 826. Most Profit Assigning Work/Solution.py
class Solution:
    def maxProfitAssignment(self, difficulty: List[int], profit: List[int], worker: List[int]) -> int:
        worker.sort(reverse = True)
        value = sorted([(x,y) for x,y in zip(difficulty, profit)], reverse=True, key = lambda x : x[1])
        max_profit = idx = 0
        for w in worker:
            while idx < len(value):
                if value[idx][0] <= w:
                    max_profit += value[idx][1]
                    break
                idx += 1
        return max_profit
